specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2017-04-01/servicebus.json

```diff
@@ -1144,6 +1144,230 @@
           }
         }
       }
+    },	
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/migrationConfigurations": {
+      "get": {
+        "tags": [
+          "migrationConfigurations"
+        ],
+        "operationId": "migrationConfigurations_List",
+        "description": "Gets all migrationConfigurations",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/namespaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the list of migrationConfigurations for servicebus namespace",
+            "schema": {
+              "$ref": "#/definitions/MigrationConfigListResult"
+            }
+          },
+          "default": {
+            "description": "ServiceBus error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/migrationConfigurations/$default": {
```

Comment:
I understand that `$default` is the only valid value now but right thing to do here is model it as a parameter with single-value enum instead of hardcoding.


```
/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}
/providers/Microsoft.ServiceBus/namespaces/{namespaceName}
/migrationConfigurations/{configName}
```


and model {configName} as:

```
{
    "name": "configName",
    "in": "path",
    "description": "The configuration name. Should always be \"$default\".",
    "required": true,
    "type": "string",
    "enum": [
       "$default"
    ],
    "x-ms-enum": {
        "name": "MigrationConfigurationName",
        "modelAsString": true
    }
}
```

Reply:
you can also move this definition to `parameters` section and refer it from eligible operations in migrationConfigurations operation group

Reply:
Since this is an enum with `ONE value` ($default) and is `required` there will not be an enum type generated in SDK instead method impl will uses a const with value $default, hence it is not exposed to user i.e.  generated methods will not have a "configName" parameter, which is exactly what you want here.

Reply:
resolved